api/lib/source/polkadotV0-source.js

```diff
@@ -529,6 +532,151 @@ class polkadotAPI {
       }
     )
   }
+
+  async getAllProposals() {
+    const api = await this.getAPI()
+
+    const [
+      blockHeight,
+      totalIssuance,
+      democracyProposals,
+      democracyReferendums,
+      treasuryProposals,
+      councilProposals,
+      councilMembers
+    ] = await Promise.all([
+      this.getBlockHeight(),
+      api.query.balances.totalIssuance(),
+      api.derive.democracy.proposals(),
+      api.derive.democracy.referendums(),
+      api.derive.treasury.proposals(),
+      api.derive.council.proposals(),
+      api.query.council.members()
+    ])
+    const allProposals = democracyProposals
+      .map((proposal) => {
+        return this.reducers.democracyProposalReducer(
+          this.network,
+          proposal,
+          totalIssuance,
+          blockHeight
+        )
+      })
+      .concat(
+        democracyReferendums.map((proposal) => {
+          return this.reducers.democracyReferendumReducer(
+            this.network,
+            proposal,
+            totalIssuance,
+            blockHeight
+          )
+        })
+      )
+      .concat(
+        treasuryProposals.proposals.map((proposal) => {
+          return this.reducers.treasuryProposalReducer(
+            this.network,
+            proposal,
+            councilMembers
+          )
+        })
+      )
+      .concat(
+        councilProposals.map((proposal) => {
+          return this.reducers.councilProposalReducer(
+            this.network,
+            proposal,
+            councilMembers,
+            blockHeight
+          )
+        })
+      )
+
+    return orderBy(allProposals, 'id', 'desc')
+  }
+
+  getProposalById() {}
+
+  getDelegatorVote() {}
+
+  async getTotalActiveAccounts() {
+    const api = await this.getAPI()
+    const accountKeys = await api.query.system.account.keys()
+    const accounts = accountKeys.map((key) => key.args[0].toHuman())
+    return accounts.length || 0
+  }
+
+  async getRecentProposals() {
+    // while we don't have proposals in DB this is the only way
+    const proposals = await this.getAllProposals()
+    // the problem in Substrate is that right now we don't have all creationTimes yet
+    return proposals.sort(
+      (a, b) => new Date(b.creationTime) - new Date(a.creationTime)
+    )
+  }
+
+  async getTopVoters() {
+    // in Substrate we simply return council members
+    const api = await this.getAPI()
+    const councilMembers = await api.query.council.members()
+    const councilMembersWithStake = await Promise.all(
```

Comment:
how about instead mapping to our validators object and sorting by voting power?

Reply:
I followed the specs in the issue. I think it depends on the voting type that council members vote are more important? Like in treasury proposals for example, while in referendums it doesn't make a difference. But will need to research more

Reply:
Also, it doesn't seem that in Polkadot validators necessarily have more voting power? I mean, each DOT holder votingPower depends on their stake. I think validators cannot vote on behalf of their stakers

Reply:
I think you misunderstood me:
You are requerying the account data for each council member here. if all council members are validators, you can just lookup the staked balance from the validator and do not need to query again.

Reply:
yeah, but not all of them are. Look Gav for example. It is safer to query for council members directly

Reply:
ok, if that is the case then this way is safer

Reply:
I guess we still need to cache the data. at least put a caching for the query for 10 minutes or so

Reply:
Done